### PR TITLE
KEP-4222: Bump github.com/fxamacker/cbor/v2.

### PR DESCRIPTION
**Pull request by benluddy**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This release makes a number of behaviors configurable in ways that are required for CBOR support in Kubernetes.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://kep.k8s.io/4222
```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode.go [hunk 1]
```diff
@@ -84,6 +84,10 @@ var Decode cbor.DecMode = func() cbor.DecMode {
 		// When decoding an unrecognized tag to interface{}, return the decoded tag content
 		// instead of the default, a cbor.Tag representing a (number, content) pair.
 		UnrecognizedTagToAny: cbor.UnrecognizedTagContentToAny,
+
+		// For parity with JSON, strings can be decoded into time.Time if they are RFC 3339
+		// timestamps.
+		ByteStringToTime: cbor.ByteStringToTimeAllowed,
```
Comment (sttts) [inline staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode.go:90]: only code change

**Comment by sttts** [comment]
/lgtm
/approve

Only used in unreleased alpha code.

File: go.mod [hunk 2]
```diff
@@ -146,7 +146,7 @@ require (
 	github.com/exponent-io/jsonpath v0.0.0-20151013193312-d6023ce2651d // indirect
 	github.com/fatih/camelcase v1.0.0 // indirect
 	github.com/felixge/httpsnoop v1.0.3 // indirect
-	github.com/fxamacker/cbor/v2 v2.6.0 // indirect
+	github.com/fxamacker/cbor/v2 v2.7.0-beta // indirect
```
Comment (dims) [inline go.mod:149]: Do we know when v2.7.0-beta will go GA?

**Comment by dims** [comment]
looking at https://github.com/fxamacker/cbor/compare/v2.6.0...v2.7.0-beta 

@benluddy are the things you hinted in `behaviors configurable in ways that are required for CBOR support in Kubernetes` a follow up PR to this one?

**Comment by benluddy** [comment]
> looking at [fxamacker/cbor@v2.6.0...v2.7.0-beta](https://github.com/fxamacker/cbor/compare/v2.6.0...v2.7.0-beta)
> 
> @benluddy are the things you hinted in `behaviors configurable in ways that are required for CBOR support in Kubernetes` a follow up PR to this one?

Yes, there are many follow-ups for this bump. Many of them are in a single draft PR right now (https://github.com/kubernetes/kubernetes/pull/124778/commits), plus or minus some test cases. There will be at least one follow-up PR per commit in that PR.

File: go.mod (hunk 2 above)
Comment (benluddy) [reply go.mod:149]: Originally there was to be a release candidate tag for `v2.7.0` over the past weekend. Several refactoring changes merged yesterday, so the maintainer is allowing extra fuzzer soak time before that happens (https://github.com/fxamacker/cbor/discussions/555). I've shared the KEP and 1.31 timelines with the maintainer and expect that we will be able to bump to a plain release tag before code freeze.

**Comment by dims** [comment]
thanks for the additional info @benluddy 

/approve
/lgtm

**Pull request merged**